core/vtk/ttkScalarFieldSmoother/ttkScalarFieldSmoother.cpp

```diff
@@ -17,6 +17,9 @@ vtkStandardNewMacro(ttkScalarFieldSmoother);
 ttkScalarFieldSmoother::ttkScalarFieldSmoother() {
   this->SetNumberOfInputPorts(1);
   this->SetNumberOfOutputPorts(1);
+#ifdef TTK_ENABLE_MPI
+  hasMPISupport_ = true;
```

Comment:
I'd recommend to move that to the constructor of the *base* class when available.
in this example `core/base/scalarFieldSmoother/`